File: android-native/app/src/main/java/cloud/dcompany/erp/ui/screens/finance/ExpensePaymentPolicy.kt
package cloud.dcompany.erp.ui.screens.finance

internal data class ExpensePaymentOption(
    val value: String,
    val label: String,
)

/** Cash is explicit because it moves one selected open shift drawer. */
internal object ExpensePaymentPolicy {
    const val DefaultRail = "upi"

    val Options = listOf(
        ExpensePaymentOption("cash", "Cash paid from shift drawer"),
        ExpensePaymentOption("upi", "UPI (business account)"),
        ExpensePaymentOption("card", "Business debit card"),
        ExpensePaymentOption("bank", "Bank transfer"),
    )

    const val CashDrawerGuidance =
        "Cash paid-outs require selecting the exact open shift that supplied the cash. " +
            "The server reduces that drawer atomically when the saved expense syncs."
}

File: frontend/src/modules/finance/expense-payment-policy.ts
import type { ExpenseCreatePaymentRail } from '@/lib/erp-api';

export interface ExpensePaymentOption {
  value: ExpenseCreatePaymentRail;
  label: string;
}

export const DEFAULT_EXPENSE_PAYMENT_RAIL: ExpenseCreatePaymentRail = 'upi';

export const EXPENSE_PAYMENT_OPTIONS: readonly ExpensePaymentOption[] = [
  { value: 'cash', label: 'Cash from open shift drawer' },
  { value: 'upi', label: 'UPI (business account)' },
  { value: 'card', label: 'Business debit card' },
  { value: 'bank', label: 'Bank transfer' },
];

export const EXPENSE_CASH_DRAWER_GUIDANCE =
  'Cash paid-outs must be linked to the open shift drawer that supplied the money. ' +
  'The amount is deducted from that drawer immediately and restored only if an authorised user ' +
  'voids the expense while the same shift is still open.';
